ilastik/widgets/listView.py

```diff
@@ -61,38 +60,38 @@ def tableViewCellClicked(self, modelIndex):
         :param modelIndex:
         """
 
-    #         if (modelIndex.column() == self.model.ColumnID.Delete and
-    #             not self._table.model().flags(modelIndex) == Qt.NoItemFlags):
-    #             self._table.model().removeRow(modelIndex.row())
-    #
     def tableViewCellDoubleClicked(self, modelIndex):
         """
         Reimplement this function to get interaction when single click
         :param modelIndex:
         """
 
-    #         if modelIndex.column() == self.model.ColumnID.Color:
-    #             self._colorDialog.setBrushColor(self._table.model()[modelIndex.row()].brushColor())
-    #             self._colorDialog.setPmapColor (self._table.model()[modelIndex.row()].pmapColor())
-    #             self._colorDialog.exec_()
-    #             #print "brush color = {}".format(self._colorDialog.brushColor().name())
-    #             #print "pmap color  = {}".format(self._colorDialog.pmapColor().name())
-    #             self._table.model().setData(modelIndex, (self._colorDialog.brushColor(),
-    #                                               self._colorDialog.pmapColor ()))
-
     def rowMovedTest(self, logicalIndex, oldVisualIndex, newVisualIndex):
         logger.debug("{} {} {}".format(logicalIndex, oldVisualIndex, newVisualIndex))
 
     def _setListViewLook(self):
+        if self.model.columnCount() < 3:
```

Comment:
Looks like this class is just an extension of `ListView` that shows a different screen when there are no items in it's associated model, so putting label-specific logic here might not be appropriate.

Reply:
This is a custom function that was probably put in this ListView because two of the three inheriting sub-ListViews use it. So the original author probably wanted to de-duplicate but also still have the third sub-ListView inheriting from the same base class.

I added the check as a minimal sanity guard, but in retrospect one might as well take it out. Without it, the function would just throw further down in the scenario this check would catch...

Reply:
To revise that a bit: This might not be the best place, but understanding and refactoring the inheritance structure around boxListView, cropListView and labelListView wasn't the point of this miniproject. If you know something about it and can suggest a better place, please do.

I checked how the guis end up looking in the workflows Pixel Classification, Pixel + Object Classification, Cell Density Counting and NN (local). I could do that check more systematically as there are I think a couple more workflows that inherit this; though my intention with adding the `raise` here was to break tests so I don't have to do this manually.